File: src/Web/JavaScriptJS/patterns/model_view_presenter.js
const assert = require("node:assert/strict");
/** Keeps a passive view behind an interface driven by a presenter. */
function run() {
  const model = { load: () => ({ name: "Ada" }) };
  const view = { text: "", showName(value) { this.text = value; } };
  const presenter = { start() { view.showName(model.load().name); } };
  presenter.start();
  assert.equal(view.text, "Ada");
  return "model-view-presenter";
}
module.exports = { run };
if (require.main === module) run();

File: src/DataScience/MATLAB/model_view_presenter.m
function result = model_view_presenter
%MODEL_VIEW_PRESENTER Let a presenter mediate a passive view and model.
model = struct("count", 0);
view = struct("text", "");
[model, view] = presenter(model, view, "increment");
result = struct("count", model.count, "viewText", view.text);
end

function [model, view] = presenter(model, view, action)
if action == "increment"
    model.count = model.count + 1;
end
view.text = "count=" + string(model.count);
end
